components/renku_data_services/notebooks/blueprints.py

```diff
@@ -804,17 +809,49 @@ async def _handler(
             parsed_server_options = await self.nb_config.crc_validator.validate_class_storage(
                 user, resource_class_id, body.disk_storage
             )
-            work_dir = Path("/home/jovyan/work")
+            work_dir = environment.working_directory
             user_secrets: K8sUserSecrets | None = None
             # if body.user_secrets:
             #     user_secrets = K8sUserSecrets(
             #         name=server_name,
             #         user_secret_ids=body.user_secrets.user_secret_ids,
             #         mount_path=body.user_secrets.mount_path,
             #     )
-            cloud_storage: list[RCloneStorage] = []
+            cloud_storages_db = await self.storage_repo.get_storage(
+                user=user, project_id=project.id, include_secrets=True
+            )
+            cloud_storage: dict[str, RCloneStorage] = {
+                str(s.storage_id): RCloneStorage(
+                    source_path=s.source_path,
+                    mount_folder=(work_dir / s.target_path).as_posix(),
+                    configuration=s.configuration.model_dump(mode="python"),
+                    readonly=s.readonly,
+                    config=self.nb_config,
+                    name=s.name,
+                )
+                for s in cloud_storages_db
+            }
+            cloud_storage_request: dict[str, RCloneStorage] = {
+                s.storage_id: RCloneStorage(
+                    source_path=s.source_path,
+                    mount_folder=(work_dir / s.target_path).as_posix(),
+                    configuration=s.configuration,
+                    readonly=s.readonly,
+                    config=self.nb_config,
+                    name=None,
+                )
+                for s in body.cloudstorage or []
+                if s.storage_id is not None
```

Comment:
question: historically, users could mount arbitrary, unsaved cloud storage and the UI would just post a body here, in which case it wouldn't have a storage_id. I don't think this is around in v2 but v1 might still use this. So I'm not entirely sure if this if statement will break things and we'd have to generate ad-hoc ids or if this is fine?

Reply:
For this first implementation of the cloud storage in V2 lets keep things simple and not allow people to mount cloud storage that is unsaved. Then if needed we can add this functionality. This also needs some design because in theory we could allow this in many other places in the API where we currently expect something that exists in the DB. So I will make the storage ID required in the post request to launch a session. If a payload is provided then it just overrides the cloud storage saved in our DB.